qiskit_ibm_runtime/runtime_job.py

```diff
@@ -249,7 +249,11 @@ def cancel(self) -> None:
                 raise RuntimeInvalidStateError(f"Job cannot be cancelled: {ex}") from None
             raise IBMRuntimeError(f"Failed to cancel job: {ex}") from None
         self.cancel_result_streaming()
-        self._status = JobStatus.CANCELLED
+        self.status()
+        if self._status != JobStatus.CANCELLED:
+            raise RuntimeInvalidStateError(
+                f"Failed to cancel job: {self.job_id()} has a status of {self._status}"
+            )
```

Comment:
I think this is a bug - it appears that it's possible to call the cancel [endpoint](https://runtime-us-east.quantum-computing.ibm.com/openapi#/Jobs/cancel_job_jid) and no error is raised but the job is actually completed. 

Either way I don't think we should manually set the status to cancelled and instead actually confirm the status.